braindecode/augmentation/transforms.py

```diff
@@ -0,0 +1,1000 @@
+            random_state=random_state
+        )
+
+
+class SignFlip(Transform):
+    """ Flip the sign axis of each feature sample with a given probability
+
+    Parameters
+    ----------
+    probability : float | callable
+        Either a float between 0 and 1 defining the uniform probability of
+        applying the operation, or a `callable` (modelling a non-uniform
+        probability law for example) taking no input and returning a boolean.
+    magnitude : object, optional
+        Always ignored, exists for compatibility.
+    mag_range : object, optional
+        Always ignored, exists for compatibility.
+    random_state: int | numpy.random.Generator, optional
+        Seed to be used to instantiate numpy random number generator instance.
+        Used to decide whether or not to transform given the probability
+        argument. Defaults to None.
+    """
+
+    def __init__(
+        self,
+        probability,
+        magnitude=None,
+        mag_range=None,
+        random_state=None
+    ):
+        super().__init__(
+            operation=sign_flip,
+            probability=probability,
+            random_state=random_state
+        )
+
+
+class DownsamplingShift(Transform):
+    """ Downsamples and offsets features matrix with a given probability
+
+    Augmentation proposed in [1]_
+
+    DEPRECATED
+
+    Parameters
+    ----------
+    probability : object, optional
+        Always ignored, exists for compatibility.
+    magnitude : object, optional
+        Always ignored, exists for compatibility.
+    mag_range : object, optional
+        Always ignored, exists for compatibility.
+    factor: int, optional
+        Factor by which X should be downsampled. For example, if factor is 2,
+        only every other column of X will be kept. Defaults to 2.
+    offset: int, optional
+        Offset (in number of columns) to be used to time-shift the data.
+        Offset needs to be less than factor.
+        When downsampling by a factor N, you have N different offsets possible.
+        If not value is passed to ofset, it is randomly relected between 0 and
+        factor-1.
+    random_state: int | numpy.random.Generator, optional
+        Seed to be used to instantiate numpy random number generator instance.
+        Used to decide whether or not to transform given the probability
+        argument and to sample the offset when omitted. Defaults to None.
+
+    References
+    ----------
+    .. [1] Frydenlund, A., & Rudzicz, F. (2015). Emotional Affect Estimation
+        Using Video and EEG Data in Deep Neural Networks. Proceedings of
+        Canadian Conference on Artificial Intelligence 273-280.
+
+    """
+
+    def __init__(
+        self,
+        probability=None,
+        magnitude=None,
+        mag_range=None,
+        factor=2,
+        offset=None,
+        random_state=None
+    ):
+        assert isinstance(factor, int), "factor has to be an int"
+        kwargs = {'random_state': random_state}
```

Comment:
you could drop this `kwargs` and simply modify `random_state = self.rng` in line 136. This will be more explicit.